che-theia-factory/src/browser/factory-theia-client.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2018-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { injectable, inject } from "inversify";
+import { MessageService } from "@theia/core/lib/common";
+import { FrontendApplicationContribution, FrontendApplication } from '@theia/core/lib/browser';
+import { EnvVariablesServer, EnvVariable } from "@theia/core/lib/common/env-variables";
+import { Resources } from "./resources";
+import axios, { AxiosInstance, AxiosResponse } from 'axios';
+import { Git, Repository } from "@theia/git/lib/common";
+import { IFactory } from "./types.d"
+import { IProjectConfig } from "@eclipse-che/workspace-client";
+
+const queryString = require('query-string');
+
+@injectable()
+export class FactoryTheiaClient implements FrontendApplicationContribution {
+
+    private static axiosInstance: AxiosInstance = axios;
+
+    private envVariables:EnvVariable[]|undefined;
+
+    constructor(
+        @inject(MessageService) private readonly messageService: MessageService,
+        @inject(EnvVariablesServer) private readonly envVariablesServer: EnvVariablesServer,
+        @inject(Git) protected readonly git: Git,
+    ) { }
+
+    async onStart(app: FrontendApplication) {
+        const factoryid = queryString.parse(window.location.search)['factory-id'];
+        if (!factoryid) {
+            return;
+        }
+
+        this.envVariables = await this.envVariablesServer.getVariables();
+        if(!this.envVariables){
+            return;
+        }
+
+        const cheApiExternalVar =  this.getEnvVariable('CHE_API_EXTERNAL');
+        if (!cheApiExternalVar) {
+            return;
+        }
+
+        const projectsRootEnvVar = this.getEnvVariable('CHE_PROJECTS_ROOT');
+        var projectsRoot = "/projects";
+        if (projectsRootEnvVar && projectsRootEnvVar.value) {
+            projectsRoot = projectsRootEnvVar.value;
+        }
+
+        const factory = new Resources(FactoryTheiaClient.axiosInstance, String(cheApiExternalVar.value));
+        const response: AxiosResponse<IFactory> = await factory.getById<IFactory>(factoryid);
+
+        response.data.workspace.projects.forEach((project: IProjectConfig) => {
+            if (!project.source) {
+                return;
+            }
+
+            const source = project.source;
+            const projectPath = projectsRoot + project.path;
+
+            this.messageService.info(`Cloning ... ${source.location} to ${projectPath}...`)
```

Comment:
as per comment of Stevan, could we use a single message for all projects ? (and trying to update message for each project)

Reply:
Let's create another issue where we can discuss improving it to have a single message for all projects. The good thing with the current approach is that we can see the progress as cloning may take some times for some projects and less time for others.

Reply:
@slemeur

Reply:
It seems to me that the messages could be rationalized. I would think that:
- Start cloning project notif
- Finished cloning project notif
- Cloning progress + number of projects to clone in the status bar
would be a better approach.

Is that a lot of rework? I agree we can look at improving that later, but let's have the issue attached to the epic ;)

Reply:
To me, this is not critical. But even if it was critical, we can work on that on another issue would be easier to the team to split the work and make good estimations.